src/battle_ai_switch_items.c

```diff
@@ -228,43 +225,226 @@ static bool8 FindMonThatAbsorbsOpponentsMove(void)
+                            if (GetMonData(&party[i], MON_DATA_HP) != 0
+                                && GetMonData(&party[i], MON_DATA_SPECIES2) != SPECIES_NONE
+                                && GetMonData(&party[i], MON_DATA_SPECIES2) != SPECIES_EGG
+                                && i != gBattlerPartyIndexes[gActiveBattler]
+                                && i != gBattlerPartyIndexes[BATTLE_PARTNER(gActiveBattler)]
+                                && IsBattlerGrounded(gActiveBattler)
+                                && (GetMonAbility(&party[i]) == ABILITY_MISTY_SURGE
+                                    || GetMonAbility(&party[i]) == ABILITY_ELECTRIC_SURGE)) //Ally has Misty or Electric Surge
+                                {
+                                    *(gBattleStruct->AI_monToSwitchIntoId + BATTLE_PARTNER(gActiveBattler)) = i;
+                                    BtlController_EmitTwoReturnValues(BUFFER_B, B_ACTION_SWITCH, 0);
+                                    switchMon = FALSE;
+                                    break;
+                                }
+                        }
+                    }
+            }
+                                
+            //Check if Active Pokemon can KO opponent instead of switching
+            //Will still fall asleep, but take out opposing Pokemon first
+            if (AiExpectsToFaintPlayer())
+                switchMon = FALSE;
+
+            //Checks to see if active Pokemon can do something against sleep
+            if (monAbility == (ABILITY_NATURAL_CURE | ABILITY_SHED_SKIN | ABILITY_EARLY_BIRD)
+                || holdEffect == (HOLD_EFFECT_CURE_SLP | HOLD_EFFECT_CURE_STATUS)
+                || HasMove(gActiveBattler, MOVE_SLEEP_TALK)
+                || (HasMoveEffect(gActiveBattler, MOVE_SNORE) && AI_GetTypeEffectiveness(MOVE_SNORE, gActiveBattler, opposingBattler) >= UQ_4_12(1.0))
+                || (IsBattlerGrounded(gActiveBattler)
+                    && (HasMove(gActiveBattler, MOVE_MISTY_TERRAIN) || HasMove(gActiveBattler, MOVE_ELECTRIC_TERRAIN)))
+                )
+                switchMon = FALSE;
+
+            //Check if Active Pokemon evasion boosted and might be able to dodge until awake
+            if (gBattleMons[gActiveBattler].statStages[STAT_EVASION] > (DEFAULT_STAT_STAGE + 3)
+                && AI_DATA->abilities[opposingBattler] != ABILITY_UNAWARE
+                && AI_DATA->abilities[opposingBattler] != ABILITY_KEEN_EYE
+                && !(gBattleMons[gActiveBattler].status2 & STATUS2_FORESIGHT)
+                && !(gStatuses3[gActiveBattler] & STATUS3_MIRACLE_EYED))     
+                switchMon = FALSE;
+
+        }
+
+        //Secondary Damage
+        if (monAbility != ABILITY_MAGIC_GUARD
+            && !AiExpectsToFaintPlayer())
+        {
+            //Toxic
+            moduloChance = 2; //50%
+            if (gBattleMons[gActiveBattler].status1 & (STATUS1_TOXIC_COUNTER > 2)
```

Comment:
is this supposed to check if toxic turn count is greater than 2? I think you would need something like `(gBattleMons[gActiveBattler].status1 & STATUS1_TOXIC_COUNTER) != STATUS1_TOXIC_TURN(2))` instead

Reply:
I was wondering why this wasn't working. But yes, that is what this is supposed to be doing. Although I have modified the recommended code above to give the 50% switchout chance for each turn when toxic turn count is greater than or equal to 2.